File: src/frontend/src/data/entities/connector/categories/RepositoryConnectorCategory.ts
import { type ConnectorCategoryID } from "@common/data/entities/connector/Connector";

import { ConnectorCategory } from "./ConnectorCategory";

export class RepositoryConnectorCategory extends ConnectorCategory {
    public static readonly CategoryID: ConnectorCategoryID = "repository";

    public constructor() {
        super(
            "Repository",
            "Publish your project to make it available to the public.",
            "Publish",
            "r-connector-category-repository-tag",
            "r-connector-category-repository-button",
        );
    }
}

File: src/connectors/base/data/entities/connector/categories/repository_connector_category.py
import typing
from dataclasses import dataclass

from common.py.data.entities.connector import ConnectorCategoryID

from .connector_category import ConnectorCategory


@dataclass(frozen=True, kw_only=True)
class RepositoryConnectorCategory(ConnectorCategory):
    category_id: typing.ClassVar[ConnectorCategoryID] = "repository"

    def __init__(self):
        super().__init__(name="Repository")
